### ISSUE TITLE
Apparent data loss in shapefile conversion from geojson?

### ISSUE THREAD

**Issue by MiguelDickson** [mbloch/mapshaper#262]
Hi there! Love mapshaper, have been using it for several months now; it's a great tool both as GUI and command-line! I believe the below is a new issue, but it's possible I'm also missing some kind of default in either of the GUI or command-line tool, so my apologies if so. I've tried looking in the issues, but am having no luck finding something that seems to fit.

For my use case, I need to convert a particular dense geojson to a shapefile. That geojson has several properties, many of which are JSON objects themselves. When I import the geojson, mapshaper seems to preserve the properties properly, but when I export this data back to shapefile, I lose any properties that are not flat. I've included a simple geojson at the bottom of this issue for reproduction.

Steps: 
1) Import the below geojson.
    a) All fields present and viewable
2) Export as shapefile to a local file on disk
3) Load all four exported files from the .zip back to a new mapshaper instance
4) Expected: all fields still present and viewable
    Actual: prop_1 and prop_2 present, but 'nested_prop' is now 'null'.

Thanks again, a big fan of the tool! =D

```
{
  "type": "FeatureCollection",
  "features": [
    {
      "geometry": {
        "type": "LineString",
        "coordinates": [
          [
            -79.6727618,
            40.3220576
          ],
          [
            -79.67281343830194,
            40.32203565959857
          ]
        ]
      },
      "type": "Feature",
      "properties": {
        "nested_prop": {
          "test_1": "blah"
        },
        "prop_1": "blue",
        "prop_2": 2
      }
    }
  ]
}

**Comment by mbloch** [comment]
You're getting nulls because Mapshaper doesn't currently know how to save nested properties to a Shapefile. In order to preserve this kind of data, mapshaper would have to convert the nested properties to one of the data types supported by the Shapefile format (basically, number, string, boolean and date). 

The obvious option is for mapshaper to convert nested properties to JSON strings. Shapefile strings can't be longer than 254 characters, I think, which could work for simple data objects.

Does this sound  like a reasonable solution?

**Comment by MiguelDickson** [comment]
Hi Matthew, and thanks for the very timely response! Actually, I wasn't aware of these particular limitations of the shapefile format; in fact, I used mapshaper precisely because I've found it handy to convert things on request to shapefile and a couple of the other formats. [I had a request a few degrees downstream from me; normally I and other colleagues work with geojsons precisely because they are a very flexible/malleable format, much as I suspect is driving the popularity of JSON itself of late.]

I'm not sure I have a strong intuition/opinion re: the implicit conversion. Given that it's a limitation of the shapefile format itself, I would probably just be in favor of an explicit warning in the export process (pop-up in the GUI; some kind of console/log output in the cmd line tool), but that's just one opinion!
